generator/src/main/java/com/scottlogic/deg/generator/generation/VelocityMonitor.java

```diff
@@ -49,15 +55,42 @@ public void rowEmitted(GeneratedObject row) {
 
     @Override
     public void endGeneration() {
-        reportVelocity(0);
-        System.out.println("\n\nGeneration finished at: " + simpleDateFormat.format(new Date()));
+        timer.cancel();
+
+        LocalDateTime finished = LocalDateTime.now();
+        Duration totalDuration = Duration.between(startedGenerating, finished);
+
+        BigDecimal nanoSecondsAsMilliseconds = BigDecimal
+            .valueOf(totalDuration.getNano())
+            .divide(nanoSecondsInMillisecond, RoundingMode.DOWN);
+
+        BigDecimal secondsAsMilliseconds = BigDecimal
+            .valueOf(totalDuration.getSeconds())
+            .multiply(millisecondsInSecond);
+
+        BigDecimal totalMilliseconds = nanoSecondsAsMilliseconds.add(secondsAsMilliseconds);
+        BigInteger averageRowsPerSecond = new BigDecimal(rowsEmitted)
```

Comment:
So after talking to you I understand why some of this is necessary but is it worth commenting for clarity why we use biginteger/bigdecimal and the maths involved here?